### PR TITLE
Refactored the code according to audit report:-

### PR DESCRIPTION
**Pull request by codebuster22**
1) Created a local var in Seed.buy() to store balance of current Seed contract to avoid calling balanceOf() twice.
2) Corrected Error Statement at StakingRewards
3) Shifted state variable declaration above initialise()
4) explicit type declaration for uint256 at Seed and SeedFactory

### CODE REVIEW

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -164,17 +164,19 @@ contract Seed {
         seedRemainder = seedRemainder.sub(_seedAmount);
         //  fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
         uint256 fundingAmount = (_seedAmount.mul(price)).div(PCT_BASE);
+        // Funding Token balancec of this contract;
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:167]: Pls fix to `Funding Token balance of this contract`

File: contracts/seed/Seed.sol [hunk 2]
```diff
@@ -164,17 +164,19 @@ contract Seed {
...
+        uint256 seedContractFundingTokenBal = fundingToken.balanceOf(address(this));
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:168]: why such a long name? we can call it just `fundingBalance`

File: contracts/seed/SeedFactory.sol [hunk 3]
```diff
@@ -95,10 +95,10 @@ contract SeedFactory is CloneFactory {
     function deploySeed(
         address          _admin,
         address[] memory _tokens,
-        uint[]    memory _softAndHardCap,
-        uint  	         _price,
-        uint 	         _startTime,
-        uint 	         _endTime,
+        uint256[]    memory _softAndHardCap,
```
Comment (kinspotting) [inline contracts/seed/SeedFactory.sol:98]: pls change indent to spaces

**Review by kinspotting — CHANGES_REQUESTED** [review]
needs minor changes

File: contracts/seed/Seed.sol (hunk 1 above)
Comment (codebuster22) [reply contracts/seed/Seed.sol]: Corrected

File: contracts/seed/Seed.sol (hunk 2 above)
Comment (codebuster22) [reply contracts/seed/Seed.sol]: renamed

File: contracts/seed/SeedFactory.sol (hunk 3 above)
Comment (codebuster22) [reply contracts/seed/SeedFactory.sol]: Updated with spaces. Also how do I find if it is a tab or space?

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**